doc/howto/opam-file-generation.rst

```diff
@@ -0,0 +1,129 @@
+that appear in the ``(libraries)`` fields of your project, and put this list in
+the ``(depends)`` field of the corresponding ``(package)``. See
+:doc:`../explanation/ocaml-ecosystem` for the difference between libraries and
+packages.
+
+Example: you have a library that looks like::
+
+  (library
+   (public_name frobnitz)
+   (libraries lwt fmt))
+
+You can declare the package as::
+
+  (package
+   (name frobnitz)
+   (depends lwt fmt)
+
+Also add common metadata using ``(authors)``, ``(maintainers)``, ``(license)``,
+``(source)``, as well as a ``(synopsis)`` and a ``(description)`` for
+
+For an existing package
+^^^^^^^^^^^^^^^^^^^^^^^
+
+If you already have an opam file (or several of them), you can convert it by
+following the rules in :ref:`package`.
+
+For example, if your opam file looks like:
+
+.. code::
+
+   opam-version: 2.0
+   authors: ["Anil Madhavapeddy" "Rudi Grinberg"]
+   maintainers: ["team@mirage.org"]
+   name: "cohttp-async"
+   synopsis: "HTTP client and server for the Async library"
+   description: "A _really_ long description"
+   license: "ISC"
+   bug-reports: "https://github.com/mirage/ocaml-cohttp/issues"
+   homepage: "https://github.com/mirage/ocaml-cohttp/"
+   dev-repo: "git+https://github.com/mirage/ocaml-cohttp.git"
+   build: [
+     ["dune" "subst"] {dev}
+     [
+       "dune"
+       "build"
+       "-p"
+       name
+       "-j"
+       jobs
+       "@install"
+       "@runtest" {with-test}
+       "@doc" {with-doc}
+     ]
+   ]
+   depends: [
+     "dune" { >= "3.4" }
+     "odoc" { with-doc }
+     "cohttp" { >= "1.0.2" }
+     "conduit-async" { >= "1.0.3" }
+     "async" { >= "v0.10.0" }
+   ]
+
+You can express this as::
+
+   (source (github mirage/ocaml-cohttp))
+   (license ISC)
+   (authors "Anil Madhavapeddy" "Rudi Grinberg")
+   (maintainers "team@mirage.org")
+
+   (package
+    (name cohttp-async)
+    (synopsis "HTTP client and server for the Async library")
+    (description "A _really_ long description")
+    (depends
+     (cohttp (>= 1.0.2))
+     (conduit-async (>= 1.0.3))
+     (async (>= v0.10.0))))
+
+General notes and tips
+^^^^^^^^^^^^^^^^^^^^^^
+
+- do not declare a dependency on the ``dune`` and ``odoc`` packages. Dune will
+  generate them with the right constraints.
+- for fields that are common between packages (like ``(authors)`` or
+  ``(license)``), you can use a global one rather than replicate it between
+  packages.
+- if you use a platform such as GitHub you can use ``(source)`` as a shorthand
+  instead of specifying ``(bug_reports)``, ``(homepage)``, etc.
+- ``(package)`` stanzas do not support all opam fields or complete syntax for
+  dependency specifications. If the package you are adapting requires this,
+  keep the corresponding opam fields in a ``pkg.opam.template`` file. See
+  :doc:`../reference/packages`.
+- it is not necessary to specify ``(version)``, this will be added at release
+  time if you use `dune-release`_.
```

Comment:
```suggestion
- Do not declare a dependency on the ``dune`` and ``odoc`` packages. Dune will
  generate them with the right constraints.
- For fields that are common between packages (like ``(authors)`` or
  ``(license)``), you can use a global one rather than replicate it between
  packages.
- If you use a platform such as GitHub you can use ``(source)`` as a shorthand
  instead of specifying ``(bug_reports)``, ``(homepage)``, etc.
- ``(package)`` stanzas do not support all opam fields or complete syntax for
  dependency specifications. If the package you are adapting requires this,
  keep the corresponding opam fields in a ``pkg.opam.template`` file. See
  :doc:`../reference/packages`.
- It is not necessary to specify ``(version)``, this will be added at release
  time if you use `dune-release`_.
```